packages/braintree-integration/src/braintree-fastlane/braintree-fastlane-utils.ts

```diff
@@ -0,0 +1,541 @@
+    BraintreeInitializationData,
+    BraintreeIntegrationService,
+} from '@bigcommerce/checkout-sdk/braintree-utils';
+import {
+    CardInstrument,
+    CustomerAddress,
+    InvalidArgumentError,
+    MissingDataError,
+    MissingDataErrorType,
+    PaymentIntegrationService,
+    PaymentMethodClientUnavailableError,
+    UntrustedShippingCardVerificationType,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+import { BrowserStorage } from '@bigcommerce/checkout-sdk/storage';
+
+export default class BraintreeFastlaneUtils {
+    private braintreeConnect?: BraintreeConnect;
+    private braintreeFastlane?: BraintreeFastlane;
+    private methodId?: string;
+
+    constructor(
+        private paymentIntegrationService: PaymentIntegrationService,
+        private braintreeIntegrationService: BraintreeIntegrationService,
+        private browserStorage: BrowserStorage,
+    ) {}
+
+    async getDeviceSessionId(): Promise<string | undefined> {
+        const cart = this.paymentIntegrationService.getState().getCart();
+
+        return this.braintreeIntegrationService.getSessionId(cart?.id);
+    }
+
+    /**
+     *
+     * Initialization method
+     *
+     */
+    async initializeBraintreeConnectOrThrow(
+        methodId: string,
+        styles?: BraintreeConnectStylesOption,
+    ) {
+        const state = this.paymentIntegrationService.getState();
+        const cart = state.getCart();
+        const storeConfig = state.getStoreConfigOrThrow();
+        const { clientToken, config, initializationData } =
+            state.getPaymentMethodOrThrow<BraintreeInitializationData>(methodId);
+
+        if (!clientToken || !initializationData) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        this.methodId = methodId;
+
+        this.braintreeIntegrationService.initialize(clientToken, storeConfig);
+        this.braintreeConnect = await this.braintreeIntegrationService.getBraintreeConnect(
+            cart?.id,
+            config.testMode,
+            styles,
+        );
+    }
+
+    async initializeBraintreeFastlaneOrThrow(
+        methodId: string,
+        styles?: BraintreeFastlaneStylesOption,
+    ) {
+        const state = this.paymentIntegrationService.getState();
+        const cart = state.getCart();
+        const storeConfig = state.getStoreConfigOrThrow();
+        const { clientToken, config, initializationData } =
+            state.getPaymentMethodOrThrow<BraintreeInitializationData>(methodId);
+
+        if (!clientToken || !initializationData) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        this.methodId = methodId;
+
+        this.braintreeIntegrationService.initialize(clientToken, storeConfig);
+        this.braintreeFastlane = await this.braintreeIntegrationService.getBraintreeFastlane(
+            cart?.id,
+            config.testMode,
+            styles,
+        );
+    }
```

Comment:
It might be ok to combine this 2 methods and set `this.braintreeFastlane` or `this.braintreeConnect` based on `isFastlaneEnabled` from payment initializationData